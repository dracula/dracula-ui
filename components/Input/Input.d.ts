import React, { HTMLAttributes } from 'react';
import { ColorMap } from '../../base/colors';
import { MarginMixin, PaddingMixin } from '../../base/spacing';
export declare const inputVariants: {
    normal: string;
    outline: string;
};
export declare const inputSizes: {
    large: string;
    medium: string;
    small: string;
};
export declare const inputColors: Partial<ColorMap>;
/** Input Props */
export interface InputProps extends HTMLAttributes<HTMLInputElement>, PaddingMixin, MarginMixin {
    /**
     * The Dracula UI theme color to be used
     */
    color?: keyof typeof inputColors;
    /**
     * Controls the size of the input based on pre-configured Dracula UI sizes.
     */
    size?: keyof typeof inputSizes;
    /**
     * Controls the variation the input.
     * `normal` -> Regular Input component with a light background color.
     * `outline` -> Keeps the accent color, but removes the background.
     */
    variant?: keyof typeof inputVariants;
}
/**
 * Input is a styled HTML Input.
 *
 * There are no behavior changes applied to the native HTML tag other
 * than light styling done via CSS in order to keep inputs accessible.
 */
export declare const Input: React.FC<InputProps>;
