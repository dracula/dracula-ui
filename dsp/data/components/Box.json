{
  "class": "component",
  "type": "page",
  "id": "Box",
  "name": "Box",
  "last_updated": "2021-03-14T00:08:30.329Z",
  "last_updated_by": "System",
  "description": "Box is the most primitive component of Dracula UI.\nUsing Box allows for consumers of the components library to compose\nmore complex patterns, components, and UIs.\n\nBox includes built-in Color and Spacing properties that make building\ncomplex components convenient and consistent.",
  "related_entity_ids": [],
  "tags": ["component"],
  "snippets": {
    "trigger": "box",
    "languages": {
      "javascriptreact": "<Box color=\"purple\">Dracula</Box>;\n",
      "html": "<div color=\"purple\" class=\"drac-box drac-bg-purple\">Dracula</div>\n"
    }
  },
  "ext_com_draculaui_variations": [
    {
      "title": "Usage",
      "html": "<div color=\"purple\" class=\"drac-box drac-bg-purple\">Dracula</div>\n",
      "react": "<Box color=\"purple\">Dracula</Box>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/Box.png",
      "svgData": "./dsp/assets/svgs/Box.svg"
    },
    {
      "title": "Colors",
      "html": "<div color=\"pinkPurple\" class=\"drac-box drac-bg-pink-purple\">Pink Purple</div>\n",
      "react": "<Box color=\"pinkPurple\">Pink Purple</Box>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxColors.png",
      "svgData": "./dsp/assets/svgs/BoxColors.svg"
    },
    {
      "title": "Rounded",
      "html": "<div style=\"display: flex; flex-direction: row\">\n  <div\n    color=\"purple\"\n    rounded=\"none\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-none\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"sm\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-sm\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"base\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"lg\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-lg\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"2xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-2xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"3xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-3xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"full\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-full\"\n  ></div>\n</div>\n",
      "react": "<div style={{ display: \"flex\", flexDirection: \"row\" }}>\n  <Box\n    color=\"purple\"\n    rounded=\"none\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"sm\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"base\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"lg\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"2xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"3xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"full\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxRounded.png",
      "svgData": "./dsp/assets/svgs/BoxRounded.svg"
    },
    {
      "title": "Generic Elements",
      "html": "<div>\n  <button as=\"button\" color=\"pinkPurple\" class=\"drac-box drac-bg-pink-purple\">\n    Button</button\n  ><a as=\"a\" color=\"green\" href=\"#\" class=\"drac-box drac-bg-green\">Link</a>\n</div>\n",
      "react": "<div>\n  <Box as=\"button\" color=\"pinkPurple\">\n    Button\n  </Box>\n  <Box as=\"a\" color=\"green\" href=\"#\">\n    Link\n  </Box>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxGeneric Elements.png",
      "svgData": "./dsp/assets/svgs/BoxGeneric Elements.svg"
    }
  ],
  "ext_com_draculaui_props": {
    "color": {
      "defaultValue": null,
      "description": "The background color.",
      "name": "color",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": {
        "name": "\"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
      }
    },
    "rounded": {
      "defaultValue": null,
      "description": "The border radius.",
      "name": "rounded",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"base\" | \"lg\" | \"xl\" | \"2xl\" | \"3xl\" | \"full\" | undefined"
      }
    },
    "spacing": {
      "defaultValue": null,
      "description": "Dracula UI standard spacing properties.",
      "name": "spacing",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "SpacingPropType | undefined" }
    },
    "margin": {
      "defaultValue": null,
      "description": "Dracula UI standard margin properties.",
      "name": "margin",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "MarginPropType | undefined" }
    },
    "as": {
      "defaultValue": null,
      "description": "The HTML element to be used",
      "name": "as",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
    }
  },
  "ext_com_draculaui_docgen": {
    "tags": {},
    "description": "Box is the most primitive component of Dracula UI.\nUsing Box allows for consumers of the components library to compose\nmore complex patterns, components, and UIs.\n\nBox includes built-in Color and Spacing properties that make building\ncomplex components convenient and consistent.",
    "displayName": "Box",
    "methods": [],
    "props": {
      "color": {
        "defaultValue": null,
        "description": "The background color.",
        "name": "color",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": {
          "name": "\"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
        }
      },
      "rounded": {
        "defaultValue": null,
        "description": "The border radius.",
        "name": "rounded",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"base\" | \"lg\" | \"xl\" | \"2xl\" | \"3xl\" | \"full\" | undefined"
        }
      },
      "spacing": {
        "defaultValue": null,
        "description": "Dracula UI standard spacing properties.",
        "name": "spacing",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "SpacingPropType | undefined" }
      },
      "margin": {
        "defaultValue": null,
        "description": "Dracula UI standard margin properties.",
        "name": "margin",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "MarginPropType | undefined" }
      },
      "as": {
        "defaultValue": null,
        "description": "The HTML element to be used",
        "name": "as",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
      }
    }
  }
}
