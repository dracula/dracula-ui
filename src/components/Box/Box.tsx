import cx from 'classnames/dedupe'
import React, { AllHTMLAttributes } from 'react'
import { borderColors, colors, glowColors } from '../../base/colors'
import {
  PaddingMixin,
  paddingMixin,
  marginMixin,
  MarginMixin
} from '../../base/spacing'

type Element = HTMLElementTagNameMap

export const roundedBorders = {
  none: 'drac-rounded-none',
  sm: 'drac-rounded-sm',
  base: 'drac-rounded',
  lg: 'drac-rounded-lg',
  xl: 'drac-rounded-xl',
  '2xl': 'drac-rounded-2xl',
  '3xl': 'drac-rounded-3xl',
  full: 'drac-rounded-full'
}

/**
 * Box Props
 */
export type BoxProps<K extends keyof Element = 'div'> = {
  /** The background color. */
  color?: keyof typeof colors

  /** The glow color. */
  glowColor?: keyof typeof glowColors

  /** The border color. */
  borderColor?: keyof typeof borderColors

  /** The border radius. */
  rounded?: keyof typeof roundedBorders

  /** The HTML element to be used */
  as?: K
} & AllHTMLAttributes<K> &
  PaddingMixin &
  MarginMixin

/**
 * Box is the most primitive component of Dracula UI.
 * Using Box allows for consumers of the components library to compose
 * more complex patterns, components, and UIs.
 *
 * Box includes built-in Color and Spacing properties that make building
 * complex components convenient and consistent.
 */
export function Box<T extends keyof Element>(props: BoxProps<T>) {
  const finalProps = {
    ...props,
    className: cx(
      `drac-box`,
      props.className,
      props.color && colors[props.color],
      props.glowColor && glowColors[props.glowColor],
      props.borderColor && borderColors[props.borderColor],
      props.rounded && roundedBorders[props.rounded],
      ...paddingMixin(props),
      ...marginMixin(props)
    )
  }

  const as = finalProps.as ?? 'div'
  return React.createElement(as, finalProps, props.children)
}

Box.displayName = 'Box'
