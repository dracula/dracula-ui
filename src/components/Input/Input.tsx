import cx from 'classnames/dedupe'
import React, { InputHTMLAttributes } from 'react'
import { ColorMap } from '../../base/colors'
import {
  cleanProps,
  marginMixin,
  MarginMixin,
  paddingMixin,
  PaddingMixin
} from '../../base/spacing'

export const inputVariants = {
  normal: 'drac-input',
  outline: 'drac-input-outline'
}

export const inputSizes = {
  large: 'drac-input-lg',
  medium: 'drac-input',
  small: 'drac-input-sm'
}

export const borderSizes = {
  large: 'drac-input-border-lg',
  medium: 'drac-input-border-md',
  small: 'drac-input-border-sm'
}

export const inputColors: Partial<ColorMap> = {
  white: 'drac-input-white drac-text-white',
  cyan: 'drac-input-cyan drac-text-cyan',
  green: 'drac-input-green drac-text-green',
  orange: 'drac-input-orange drac-text-orange',
  pink: 'drac-input-pink drac-text-pink',
  purple: 'drac-input-purple drac-text-purple',
  red: 'drac-input-red drac-text-red',
  yellow: 'drac-input-yellow drac-text-yellow'
}

/** Input Props */
export interface InputProps
  extends Omit<InputHTMLAttributes<HTMLInputElement>, 'size' | 'type'>,
    PaddingMixin,
    MarginMixin {
  /**
   * The Dracula UI theme color to be used
   */
  color?: keyof typeof inputColors

  /**
   * Controls the size of the input based on pre-configured Dracula UI sizes.
   */
  size?: keyof typeof inputSizes | number
 
  /**
   * Controls the border size of the input based on pre-configured Dracula UI sizes.
   */
  'borderSize'?: keyof typeof borderSizes | number

  /**
   * Controls the variation the input.
   * `normal` -> Regular Input component with a light background color.
   * `outline` -> Keeps the accent color, but removes the background.
   */
  variant?: keyof typeof inputVariants

  /**
   * Controls the type of the input.
   */
  type?:
    | 'button'
    | 'checkbox'
    | 'color'
    | 'date'
    | 'datetime-local'
    | 'email'
    | 'file'
    | 'hidden'
    | 'image'
    | 'month'
    | 'number'
    | 'password'
    | 'radio'
    | 'range'
    | 'reset'
    | 'search'
    | 'submit'
    | 'tel'
    | 'text'
    | 'time'
    | 'url'
    | 'week'
}

/**
 * Input is a styled HTML Input.
 *
 * There are no behavior changes applied to the native HTML tag other
 * than light styling done via CSS in order to keep inputs accessible.
 */
export const Input = React.forwardRef<HTMLInputElement, InputProps>(
  (props, ref) => {
    const { color, size, borderSize, variant, ...htmlProps } = props

    const finalProps: Record<string, any> = {
      ...htmlProps,
      className: cx(
        `drac-input`,
        props.className,
        variant && inputVariants[variant],
        size && typeof size === 'string' && inputSizes[size],
        borderSize && typeof borderSize === 'string' && borderSizes[borderSize], 
        color && inputColors[color],
        ...paddingMixin(props),
        ...marginMixin(props)
      )
    }

    if (size && typeof size === 'number') {
      finalProps[size] = size
    }
    
    if (borderSize && typeof borderSize === 'number') {
      finalProps[borderSize] = borderSize
    }

    return <input ref={ref} {...cleanProps(finalProps)} />
  }
)

Input.displayName = 'Input'
