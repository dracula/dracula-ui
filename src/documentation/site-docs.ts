import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from './render-component'
import React from 'react'
import fs from 'fs-extra'
import { pretty } from './pretty'
import { getDocGen } from '../documentation/docgen/doc-generator'
import { toDSP } from '../documentation/dsp/component-generator'
import puppeteer from 'puppeteer'
import { componentScreenshot } from './screenshots/component-screenshot'
import os from 'os'

interface Documentation {
  basic: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder> | SnapshotBuilder
}

jest.setTimeout(10000)

const RECORDING = process.env.RECORDING

export function docs<T>(
  componentClass: React.FunctionComponent<T> | string,
  documentation: Documentation
) {
  const name =
    typeof componentClass === 'string'
      ? componentClass
      : componentClass.displayName ?? componentClass.name

  const dspPath = `${process.cwd()}/dsp`

  describe(`Site: ${name}`, () => {
    let browser: puppeteer.Browser

    beforeAll(async () => {
      if (!RECORDING) {
        return
      }

      let executablePath = undefined
      if (os.arch() === 'arm64') {
        executablePath = '/Applications/Chromium.app/Contents/MacOS/Chromium'
      }

      browser = await puppeteer.launch({
        headless: true,
        devtools: true,
        executablePath
      })
    })

    let examples: Record<string, ComponentExample> = {}

    afterAll(async () => {
      if (!RECORDING) {
        return
      }

      await browser.close()
      const docGen = getDocGen(name)
      const dsp = toDSP(name, examples, docGen)

      await fs.mkdirp(`${dspPath}/data/components/`).catch(() => {
        // folder already exists
      })

      const path = `${dspPath}/data/components/${name}.json`
      await fs.writeFile(path, pretty(JSON.stringify(dsp), 'json'))
    })

    test('Basic Usage', async () => {
      const variation = documentation.basic()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()

      if (RECORDING) {
        const [screenshot, svg] = await componentScreenshot(
          browser,
          snapshot,
          null,
          name
        )

        snapshot.screenshot = screenshot
        snapshot.svgData = svg
      }

      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations: SnapshotBuilder[] = ([] as SnapshotBuilder[]).concat(
      documentation.variations()
    )

    variations.forEach((variation) => {
      test(variation.title, async () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()

        if (RECORDING) {
          const [screenshot, svg] = await componentScreenshot(
            browser,
            snapshot,
            variation,
            name
          )

          snapshot.screenshot = screenshot
          snapshot.svgData = svg
        }

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
