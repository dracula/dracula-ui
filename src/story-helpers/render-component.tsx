import React from 'react'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { pretty } from './pretty'

Enzyme.configure({ adapter: new Adapter() })

export type ComponentExample = {
  html: string
  docs: string
  title: string
  react?: any
  screenshot?: string
  svgData?: string
}

export type SnapshotBuilder = {
  comp: () => React.ReactElement
  docs: string
  title: string
}

export function buildSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): SnapshotBuilder {
  return {
    title,
    comp,
    docs: docs.trim()
  }
}

export function renderSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): ComponentExample {
  let component = comp()
  const html = mount(component).html()
  let wrapped = false

  // wrap the component in a div when it's not wrapped already
  // we do this so we can get the actual component example to not be shallow rendered.
  // Shallow rendering a parent "div" gives us a nice JSX representation of the example.
  if (component.type !== 'div') {
    component = <div>{component}</div>
    wrapped = true
  }

  let react = shallow(component).debug({ verbose: true })

  if (wrapped) {
    // replace wrapper div ðŸ˜¬
    react = react
      .replace(/^<div>\n/, '')
      .replace(/\n<\/div>$/, '')
      .trim()
  }

  return {
    title,
    html: pretty(html, 'html'),
    react: pretty(react, 'typescript'),
    docs
  }
}
