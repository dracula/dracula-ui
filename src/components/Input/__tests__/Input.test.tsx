import React from 'react'
import { docs } from '@/documentation/site-docs'
import { snapshot } from '@/documentation/render-component'
import { Box } from '@/components/Box/Box'
import { Input } from '@/components/Input/Input'

docs(Input, {
  basic() {
    return snapshot('Usage', Usage, 'Styles or abstracts HTML input elements.')
  },
  variations() {
    return [
      snapshot(
        'Colors',
        Colors,
        'Inputs can be customized to use any of the Dracula UI theme colors.'
      ),
      snapshot(
        'Sizes',
        Sizes,
        'Inputs can be customize to use several different sizes.'
      ),
      snapshot(
        'Variants',
        Variants,
        'Use the `outline` variant to represent subtler text Inputs'
      ),
      snapshot(
        'Type',
        Types,
        'Use the `type` property to define what kind of HTML input you would like to use'
      )
    ]
  }
})

function Usage() {
  return <Input placeholder="Input" color="white" />
}

function Types() {
  return (
    <Box>
      <Input my="sm" color="white" placeholder="Password" type="password" />
      <Input my="sm" color="white" placeholder="Date" type="date" />
      <Input my="sm" color="white" placeholder="Email" type="email" />
      <Input my="sm" color="white" placeholder="Number" type="number" />
      <Input my="sm" color="white" placeholder="Range" type="range" />
      <Input my="sm" color="white" placeholder="Telephone" type="tel" />
    </Box>
  )
}

function Colors() {
  return (
    <Box>
      <Input color="purple" placeholder="purple" m="xs" />
      <Input color="green" placeholder="green" m="xs" />
    </Box>
  )
}

function Sizes() {
  return (
    <Box>
      <Input color="white" size="small" placeholder="small" m="xs" />
      <Input color="white" size="medium" placeholder="medium" m="xs" />
      <Input color="white" size="large" placeholder="large" m="xs" />
    </Box>
  )
}

function Variants() {
  return (
    <Box>
      <Input color="white" variant="normal" placeholder="normal" m="xs" />
      <Input color="white" variant="outline" placeholder="outline" m="xs" />
    </Box>
  )
}
