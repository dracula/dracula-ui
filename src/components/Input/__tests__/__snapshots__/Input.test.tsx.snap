// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: Input Basic Usage 1`] = `
Object {
  "docs": "Styles or abstracts HTML input elements.",
  "html": "<input
  placeholder=\\"Input\\"
  class=\\"drac-input drac-input-white drac-text-white\\"
/>
",
  "react": "<Input placeholder=\\"Input\\" color=\\"white\\" />;
",
  "title": "Usage",
}
`;

exports[`Site: Input BorderSizes 1`] = `
Object {
  "docs": "Border size can be customize to use several different sizes.",
  "html": "<div class=\\"drac-box\\">
  <input
    placeholder=\\"small border outline\\"
    class=\\"drac-input drac-input-border-sm drac-input-white drac-text-white drac-m-xs\\"
  /><input
    placeholder=\\"small border outline\\"
    class=\\"drac-input drac-input-outline drac-input-border-sm drac-input-white drac-text-white drac-m-xs\\"
  /><input
    placeholder=\\"medium border outline\\"
    class=\\"drac-input drac-input-border-md drac-input-white drac-text-white drac-m-xs\\"
  /><input
    placeholder=\\"medium border outline\\"
    class=\\"drac-input drac-input-outline drac-input-border-md drac-input-white drac-text-white drac-m-xs\\"
  /><input
    placeholder=\\"large border outline\\"
    class=\\"drac-input drac-input-border-lg drac-input-white drac-text-white drac-m-xs\\"
  /><input
    placeholder=\\"large border outline\\"
    class=\\"drac-input drac-input-outline drac-input-border-lg drac-input-white drac-text-white drac-m-xs\\"
  />
</div>
",
  "react": "<Box>
  <Input
    color=\\"white\\"
    variant=\\"normal\\"
    borderSize=\\"small\\"
    placeholder=\\"small border outline\\"
    m=\\"xs\\"
  />
  <Input
    color=\\"white\\"
    variant=\\"outline\\"
    borderSize=\\"small\\"
    placeholder=\\"small border outline\\"
    m=\\"xs\\"
  />
  <Input
    color=\\"white\\"
    variant=\\"normal\\"
    borderSize=\\"medium\\"
    placeholder=\\"medium border outline\\"
    m=\\"xs\\"
  />
  <Input
    color=\\"white\\"
    variant=\\"outline\\"
    borderSize=\\"medium\\"
    placeholder=\\"medium border outline\\"
    m=\\"xs\\"
  />
  <Input
    color=\\"white\\"
    variant=\\"normal\\"
    borderSize=\\"large\\"
    placeholder=\\"large border outline\\"
    m=\\"xs\\"
  />
  <Input
    color=\\"white\\"
    variant=\\"outline\\"
    borderSize=\\"large\\"
    placeholder=\\"large border outline\\"
    m=\\"xs\\"
  />
</Box>;
",
  "title": "BorderSizes",
}
`;

exports[`Site: Input Colors 1`] = `
Object {
  "docs": "Inputs can be customized to use any of the Dracula UI theme colors.",
  "html": "<div class=\\"drac-box\\">
  <input
    placeholder=\\"purple\\"
    class=\\"drac-input drac-input-purple drac-text-purple drac-m-xs\\"
  /><input
    placeholder=\\"green\\"
    class=\\"drac-input drac-input-green drac-text-green drac-m-xs\\"
  />
</div>
",
  "react": "<Box>
  <Input color=\\"purple\\" placeholder=\\"purple\\" m=\\"xs\\" />
  <Input color=\\"green\\" placeholder=\\"green\\" m=\\"xs\\" />
</Box>;
",
  "title": "Colors",
}
`;

exports[`Site: Input Sizes 1`] = `
Object {
  "docs": "Inputs can be customize to use several different sizes.",
  "html": "<div class=\\"drac-box\\">
  <input
    placeholder=\\"small\\"
    class=\\"drac-input drac-input-sm drac-input-white drac-text-white drac-m-xs\\"
  /><input
    placeholder=\\"medium\\"
    class=\\"drac-input drac-input-white drac-text-white drac-m-xs\\"
  /><input
    placeholder=\\"large\\"
    class=\\"drac-input drac-input-lg drac-input-white drac-text-white drac-m-xs\\"
  />
</div>
",
  "react": "<Box>
  <Input color=\\"white\\" size=\\"small\\" placeholder=\\"small\\" m=\\"xs\\" />
  <Input color=\\"white\\" size=\\"medium\\" placeholder=\\"medium\\" m=\\"xs\\" />
  <Input color=\\"white\\" size=\\"large\\" placeholder=\\"large\\" m=\\"xs\\" />
</Box>;
",
  "title": "Sizes",
}
`;

exports[`Site: Input Type 1`] = `
Object {
  "docs": "Use the \`type\` property to define what kind of HTML input you would like to use",
  "html": "<div class=\\"drac-box\\">
  <input
    placeholder=\\"Password\\"
    type=\\"password\\"
    class=\\"drac-input drac-input-white drac-text-white drac-my-sm\\"
  /><input
    placeholder=\\"Date\\"
    type=\\"date\\"
    class=\\"drac-input drac-input-white drac-text-white drac-my-sm\\"
  /><input
    placeholder=\\"Email\\"
    type=\\"email\\"
    class=\\"drac-input drac-input-white drac-text-white drac-my-sm\\"
  /><input
    placeholder=\\"Number\\"
    type=\\"number\\"
    class=\\"drac-input drac-input-white drac-text-white drac-my-sm\\"
  /><input
    placeholder=\\"Range\\"
    type=\\"range\\"
    class=\\"drac-input drac-input-white drac-text-white drac-my-sm\\"
  /><input
    placeholder=\\"Telephone\\"
    type=\\"tel\\"
    class=\\"drac-input drac-input-white drac-text-white drac-my-sm\\"
  />
</div>
",
  "react": "<Box>
  <Input my=\\"sm\\" color=\\"white\\" placeholder=\\"Password\\" type=\\"password\\" />
  <Input my=\\"sm\\" color=\\"white\\" placeholder=\\"Date\\" type=\\"date\\" />
  <Input my=\\"sm\\" color=\\"white\\" placeholder=\\"Email\\" type=\\"email\\" />
  <Input my=\\"sm\\" color=\\"white\\" placeholder=\\"Number\\" type=\\"number\\" />
  <Input my=\\"sm\\" color=\\"white\\" placeholder=\\"Range\\" type=\\"range\\" />
  <Input my=\\"sm\\" color=\\"white\\" placeholder=\\"Telephone\\" type=\\"tel\\" />
</Box>;
",
  "title": "Type",
}
`;

exports[`Site: Input Variants 1`] = `
Object {
  "docs": "Use the \`outline\` variant to represent subtler text Inputs",
  "html": "<div class=\\"drac-box\\">
  <input
    placeholder=\\"normal\\"
    class=\\"drac-input drac-input-white drac-text-white drac-m-xs\\"
  /><input
    placeholder=\\"outline\\"
    class=\\"drac-input drac-input-outline drac-input-white drac-text-white drac-m-xs\\"
  />
</div>
",
  "react": "<Box>
  <Input color=\\"white\\" variant=\\"normal\\" placeholder=\\"normal\\" m=\\"xs\\" />
  <Input color=\\"white\\" variant=\\"outline\\" placeholder=\\"outline\\" m=\\"xs\\" />
</Box>;
",
  "title": "Variants",
}
`;
